provide java serialization and refactor valueMapper

jangalinski opened the pull request:
wip

- see #660
- see #662
- see #663

zambrovski commented on extension/core/src/main/kotlin/org/camunda/community/rest/config/CamundaRestClientProperties.kt:
```diff
@@ -47,8 +48,10 @@ data class CamundaRestClientProperties(
-  val deserializeVariablesOnServer: Boolean = false
+  val deserializeVariablesOnServer: Boolean = false,
 
+  @NestedConfigurationProperty
```
Missing docs for the user.

jangalinski commented:
@rohwerj can you have a look? Story: I had to implement a customValueMapper for Java Serialzation in my customer project. Basically I moved this to the format converters and also extracted the json specific behavior and made it configurable.
